quicktill/pdrivers.py

```diff
@@ -701,6 +717,50 @@ def process_canvas(self, canvas, f):
                 f.write(escpos.ep_fullcut)
             f.flush()
 
+    def _image(self, data, width, height, f):
+        # Print a PBM bit-image
+        if width > self.dpl:
+            # Image too wide for paper
+            return
+
+        # Calculate padding required to center the image
+        padding = (self.dpl - width) // 2
+        padchars = [False] * padding
+
+        # Partition the bitmap into lines
+        lines = []
+        for line in range(height):
+            start = line * (width // 8)
+            end = start + (width // 8)
+            line = padchars.copy()
+            for byte in data[start:end]:
+                for bit in f"{int(byte):08b}":
+                    line.append(bool(int(bit)))
+            lines.append(line)
+
+        # Compact up to twenty-four bit-lines into each row
+        rows = []
+        for chunk in range(height // 24):
+            group_start = chunk * 24
+            group_end = group_start + 24
+            row = []
+            for column in zip(*lines[group_start:group_end]):
+                segments = column[0:8], column[8:16], column[16:24]
+                for segment in segments:
+                    binary = str().join("1" if bit else "0" for bit in segment)
+                    row.append(int(binary, base=2))
+            rows.append(bytes(row))
+
+        # Write the commands to render the padded image
+        f.write(escpos.ep_unidirectional_on)
+        for row in rows:
+            width_info = (len(row) // 3).to_bytes(length=2, byteorder="little")
+            f.write(escpos.ep_bitimage_dd_v24 + width_info + row)
+        f.write(escpos.ep_unidirectional_off)
+
+        # Clear the line for subsequent content
+        f.write(escpos.ep_short_feed)
```

Comment:
This could be causing problems — `ep_short_feed` doesn't feed a full row height; in the QR-code printing method we are only using 6 out of 8 pins, so it only feeds 6/8 of a row I think. You'll need a different command to feed exactly 24 pins worth.